src/canisters/frontend/ic-asset/src/asset_config.rs

```diff
@@ -258,6 +251,126 @@ impl AssetConfig {
     }
 }
 
+/// This module contains various utilities needed for serialization/deserialization
+/// and pretty-printing of the `AssetConfigRule` data structure.
+mod rule_utils {
```

Comment:
In the future, I'd like to suggest submitting refactors like this in advance, separately, in order to minimize diffs and make it easier for both submitter and reviewer to see what's changed, versus what's being moved around.